### PR TITLE
Namespace colours and usecases with a slash in their name.

### PR DESCRIPTION
**Pull request by notlee**
Setting a colour or colour usecase with colour name and project name
works well with o-colors, but is not great if you  want to pass a
custom colour name to a component like o-buttons. To define two colours
four arguments must be given (the colour name and where to get the colour
from).

Instead namespaced colours and usecases using a forward slash
e.g. `o-social/twitter`.

```scss
// now
@include oColorsSetColor('o-social', 'twitter', blue);
@include oColorsSetUseCase('o-social', 'twitter-icon', (
   'background': ('color': 'twitter', 'from': 'o-social'),
   'text': 'white'
));
@include oButtonsAddTheme(
   $name: 'twitter',
   $opts: (
       'color': ('color': 'twitter', 'from': 'o-social'),
       'context': 'slate'
   ),
   $types: ('primary', 'secondary')
);

// udpated
@include oColorsSetColor('o-social/twitter', blue);
@include oColorsSetUseCase('o-social/twitter-icon', (
   'background': 'o-social/twitter',
   'text': 'white'
));
@include oButtonsAddTheme(
   $name: 'twitter',
   $opts: (
       'color': 'o-social/twitter',
       'context': 'slate'
   ),
   $types: ('primary', 'secondary')
);
```

(I renamed `oColorsGetPalette` to `oColorsGetPaletteDetails` to match with similar functions)

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to 482c8de3750eb759099833c3037d89fe6e566f92
css, master: 4.87kb decrease (-0.71kb/gzip)
css, internal: 4.52kb decrease (-0.71kb/gzip)
css, whitelabel: 4.96kb decrease (-0.86kb/gzip)

File: src/scss/_mixins.scss [hunk 1]
```diff
@@ -63,88 +58,92 @@
...
+			'only be set to an o-colors colour name such as "paper".';
+		}
+		@if(not _oColorsNameExists($color)) {
+			@error 'The "#{$usecase}" usecase property "#{inspect($property)}" ' +
+			'cannot be set to "#{inspect($color)}". The colour "#{inspect($color)}" ' +
+			'does not exist.';
 		}
 	}
 
+	// Remove the default o-colors namespace so returned namespace names
```
Comment (notlee) [inline src/scss/_mixins.scss]:
```suggestion
	// Remove the default o-colors namespace so returned color names
```

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to ab7c4d980ad25163fbcc2391340625f9956416ca
css, master: 4.87kb decrease (-0.71kb/gzip)
css, internal: 4.52kb decrease (-0.71kb/gzip)
css, whitelabel: 4.96kb decrease (-0.86kb/gzip)

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to c34243177345cdc906a964e69264ac07ab208f1a
css, master: 4.87kb decrease (-0.71kb/gzip)
css, internal: 4.52kb decrease (-0.71kb/gzip)
css, whitelabel: 4.96kb decrease (-0.86kb/gzip)

**Comment by chee** [comment]
i like this :))))

**Pull request merged**